stackdriver/storage.go

```diff
@@ -72,7 +73,30 @@ func (s *Storage) Close() error {
 // ApplyConfig updates the state as the new config requires.
 func (s *Storage) ApplyConfig(conf *config.Config) error {
 	var projectId string
-	if value, ok := conf.GlobalConfig.ExternalLabels[ProjectIdLabel]; !ok {
+	var location string
+	var clusterName string
+	el := conf.GlobalConfig.ExternalLabels
+	if md.OnGCE() {
+		if id, err := md.ProjectID(); err == nil {
```

Comment:
In order to avoid unnecessary latency, it seems like checking for an empty value before reaching out to the metadata server could result in a better user experience. Switching this condition with the one on line 82, same throughout.

Reply:
@bmoyles0117 what latency are we talking about? This happens once during startup. We should favor readability if latency is no higher than a couple seconds.

Reply:
Let me remove latency from the equation here, the lookup is only necessary if the value is empty. The way the code exists now, it will perform a metadata lookup, regardless of whether or not the value ends up being used. We should first check if it's even necessary, and when it's necessary, set the value according to the response from the metadata server.

Reply:
Done.